src/test/looping_control_test.cpp

```diff
@@ -712,6 +718,45 @@ TEST_F(LoopingControlTest, BeatLoopSize_IsSetByNumberedControl) {
     EXPECT_EQ(2.0, m_pBeatLoopSize->get());
 }
 
+TEST_F(LoopingControlTest, BeatLoopSize_SetRangeCheck) {
+    m_pBeatLoopSize->set(512.0);
+    EXPECT_EQ(512, m_pBeatLoopSize->get());
+
+    m_pBeatLoopSize->set(150.0);
+    EXPECT_EQ(150, m_pBeatLoopSize->get());
+
+    m_pBeatLoopSize->set(513.0);
+    EXPECT_EQ(150, m_pBeatLoopSize->get());
+
+    m_pButtonLoopDouble->set(1.0);
+    m_pButtonLoopDouble->set(0.0);
+    EXPECT_EQ(300.0, m_pBeatLoopSize->get());
+
+    m_pButtonLoopDouble->set(1.0);
+    m_pButtonLoopDouble->set(0.0);
+    EXPECT_EQ(300.0, m_pBeatLoopSize->get());
```

Comment:
please mention the beatloop/loopjump limits here and below

Reply:
Done